addons/stock/static/src/widgets/generate_serial.js

```diff
@@ -33,25 +33,46 @@ export class GenerateDialog extends Component {
         this.keepLines = useRef('keepLines');
         this.lots = useRef('lots');
         this.orm = useService("orm");
+        this.state = useState({
+            showGenerateButton: false,
+            cachedSerialNumber: null
+        });
         onWillStart(async () => {
             this.displayUOM = await user.hasGroup("uom.group_uom");
         });
-        onMounted(() => {
+        onMounted(async () => {
             if (this.props.mode === 'generate') {
                 this.nextSerialCount.el.value = this.props.move.data.product_uom_qty || 2;
                 if (this.props.move.data.has_tracking === 'lot') {
                     this.totalReceived.el.value = this.props.move.data.quantity;
                 }
+                // Automatically generate first serial number
+                await this._onGenerateCustomSerial();
             }
         });
     }
+
+    setCachedSerialNumber(){
+        this.nextSerial.el.value = this.state.cachedSerialNumber
+        this.state.showGenerateButton = false
+    }
+
+    onSerialInputChange(ev) {
+        const value = ev.target.value.trim();
+        if(!value || value.trim() === ''){
+            this.state.showGenerateButton = true
+        }
+    }
```

Comment:
```suggestion
    onNextSerialInputChange(ev) {
        const value = ev.target.value.trim();
        this.state.showGenerateButton = value === '';
    }
```

I would prefer renaming the method to `onNextSerialInputChange` for better clarity and consistency with the field name. WDYT?

Since the `value is already trimmed`, there’s no need to call `trim()` again inside the condition.

Also, this one handles the case where the user enters any `lot/serial number`; in that case, the `Generate button should be hidden` to maintain `consistency`.